autoprecompiles/src/memory_optimizer.rs

```diff
@@ -71,24 +62,24 @@ pub fn check_register_operation_consistency<T: FieldElement, V: Clone + Ord + Di
 }
 
 #[derive(Clone, Debug)]
-enum MemoryOp {
-    Send,
-    Receive,
+/// The type of the memory bus interaction.
+pub enum MemoryOp {
+    /// Get the previous value from memory.
+    GetPrevious,
+    /// Set the new value in memory.
+    SetNew,
 }
 
-#[derive(Clone, Debug)]
-struct MemoryBusInteraction<T: FieldElement, V> {
-    op: MemoryOp,
-    address_space: T,
-    addr: GroupedExpression<T, V>,
-    data: Vec<GroupedExpression<T, V>>,
-    #[allow(dead_code)]
-    // TODO: The timestamp is currently ignored. At some point, we should use it
-    // to assert that the bus interactions are in the right order.
-    timestamp: GroupedExpression<T, V>,
```

Comment:
I removed this item, also from `OpenVmMemoryBusInteraction`. I can't think of an easy way to actually assert this.